WordPress/src/main/java/org/wordpress/android/ui/jetpack/backup/BackupDownloadListItemState.kt

```diff
@@ -0,0 +1,88 @@
+import org.wordpress.android.ui.jetpack.backup.BackupDownloadListItemState.ViewType.HEADER
+import org.wordpress.android.ui.jetpack.backup.BackupDownloadListItemState.ViewType.SUBHEADER
+import org.wordpress.android.ui.utils.UiString
+import org.wordpress.android.ui.utils.UiString.UiStringRes
+import org.wordpress.android.ui.utils.UiString.UiStringResWithParams
+
+// todo: annmarie - tighten this up as the remainder of backup download is implemented
+sealed class BackupDownloadListItemState(val type: ViewType) {
+    open fun longId(): Long = hashCode().toLong()
+
+    sealed class HeaderState : BackupDownloadListItemState(HEADER) {
+        abstract val text: UiString
+
+        data class DetailsHeaderState(
+            override val text: UiString = UiStringRes(R.string.backup_download_details_header)
+        ) : HeaderState()
+    }
+
+    sealed class SubHeaderState : BackupDownloadListItemState(SUBHEADER) {
+        abstract val text: UiString
+
+        data class DetailsSubHeaderState(
+            override val text: UiString = UiStringRes(R.string.backup_download_details_choose_items_header)
+        ) : SubHeaderState()
+    }
+
+    sealed class DescriptionState : BackupDownloadListItemState(DESCRIPTION) {
+        abstract val text: UiString
+
+        data class DetailsSubHeaderState(
+            override val text: UiString
+        ) : DescriptionState() {
+            constructor(params: List<UiString>) : this(
+                    text = UiStringResWithParams(
+                            R.string.backup_download_details_choose_items_header,
+                            params
+                    )
+            )
+        }
+    }
+
+    sealed class IconState : BackupDownloadListItemState(DESCRIPTION) {
+        abstract val icon: Int
+        abstract val contentDescription: UiString
+
+        data class DetailsIconState(
+            @DrawableRes override val icon: Int = R.drawable.ic_get_app_24dp,
+            override val contentDescription: UiString =
+                    UiStringRes(R.string.backup_download_details_icon_content_description)
+        ) : IconState()
+    }
+
+    sealed class ActionButtonState : BackupDownloadListItemState(ACTION_BUTTON) {
+        abstract val text: UiString
+        abstract val contentDescription: UiString
+
+        data class DetailsActionButtonState(
+            override val text: UiString = UiStringRes(R.string.backup_download_details_action_button),
+            override val contentDescription: UiString =
+                    UiStringRes(R.string.backup_download_details_action_button_content_description),
+            val onClick: () -> Unit
+        ) : ActionButtonState()
+    }
+
+    data class CheckboxState(
+        val availableItemType: BackupAvailableItemType,
+        val label: UiString,
+        val checked: Boolean = false,
+        val onClick: (() -> Unit)
+    ) : BackupDownloadListItemState(CHECKBOX)
+
+    enum class ViewType(val id: Int) {
```

Comment:
> Are the enums ordered based on appearance on UI?

Yep, because I created them while I was looking at the design mocks. :) I'm not married to them either way, I'll switch them to match in a future PR.